src/pint/scripts/event_optimize.py

```diff
@@ -758,6 +758,13 @@ def main(argv=None):
 
     import emcee
 
+    # Setting up a backend to save the chains into an h5 file
+    try:
+        backend = emcee.backends.HDFBackend(ftr.model.PSR.value + "_chains.h5")
```

Comment:
It would be better to allow the user to specify a location for the (potentially large) HDF5 file - putting it in the current directory might not be helpful.

Reply:
I have added in a backend filepath argument that lets the user specify a file path for the HDF5 file